pylitterbot/robot/litterrobot4.py

```diff
@@ -233,7 +239,9 @@ def is_drawer_full_indicator_triggered(self) -> bool:
 
     @property
     def is_hopper_removed(self) -> bool | None:
-        """Return `True` if the hopper is removed."""
+        """Return `True` if the hopper is removed/disabled."""
+        if self._data.get("isHopperRemoved") is None:
+            return None
```

Comment:
Arguably this introduces a breaking change. Although the mypy signature says a `None` could be returned, it would always be `True` or `False`. I spotted this when writing tests. I'm open to reverting this.

Reply:
While technically the signature does allow for `None`, what is the purpose of introducing the return of a `None` value here for enabling/disabling the litter hopper?

Reply:
You can have the hopper accessory physically installed into your LR4, but disabled, this will have a `isHopperRemoved` of `true`. If you don't have the hopper accessory at all, it'll return `None` (I'm pretty sure).

Reply:
What I've seen before from Whisker with some of these statuses that rely on an initial trigger/change/update is that they can never go back to being in a null/non-existent state once it has been changed. I saw something similar with, I believe, the globe light level. The question that remains here is, if you completely remove the hopper, will it ever go back to a null state? If not, then I don't see the point in changing this to return None.

Would you be willing to disconnect your hopper and see what hopperStatus and isHopperRemoved change to? I don't know if that's an easy process to do so or not, but I'm curious if they "reset".

Reply:
I've removed the actually hopper attachment from the LR4, but the bonnet made for the hopper and little chute channel remain installed since that's non-trivial.

Interestingly the API is simply reporting `"hopperStatus": "ENABLED"` and `"isHopperRemoved": false`, even after 5 minutes. Their app suggests everything is nominal too, so no surprise difference there.

Even after power cycling the unit, it's not changed those values.

So I think you're correct. It'll never go to `None` again-- unless it's a full removal back to the original bonnet hardware in which case someone can open a ticket in that rare edge case 😉

Reply:
So yeah it's a matter of preference for you if you think this method should return `True` if someone who doesn't have the hopper installed uses it. Or if they should get the "bugged" behaviour of it returning `False` despite the fact they don't have the hopper installed.